src/hooks/useBasisForm/validation/validateInput.ts

```diff
@@ -7,7 +9,7 @@ export const validateInput = (
   value: string,
   inputProps: Record<string, any>
 ) => {
-  const { optional, variant } = inputProps;
+  const { optional, variant } = inputProps as InternalInputProps;
```

Comment:
Why not type `inputProps: InternalInputProps` in the argument?

Reply:
Can't make them the correct arguments as this: https://github.com/moroshko/basis/pull/228/files#diff-ef68dd42be3eb2b1ff26a064552a8eb513f9f3ede5107c2565d1592bb390d026R6

expects the type `ValidateFn` which takes the `Record<string, any>`. In order to properly type this we would have to pass the component from `as=` from the field all the way to the `ValidateFn` which will make the types very difficult to understand.

It will also force consumers to do the same and will make having shared validation much more difficult